api/base/serializers.py

```diff
@@ -80,6 +81,18 @@ def to_internal_value(self, data):
         return super(TypeField, self).to_internal_value(data)
 
 
+class TargetTypeField(TypeField):
+    """
+    Enforces that the related resource has the correct type
+    """
+
+    def to_internal_value(self, data):
+        if self.root.Meta.target_type_ != data:
+            raise Conflict()
+        # Super call intentional to prevent target_type_ being checked against type_
+        return super(TypeField, self).to_internal_value(data)
```

Comment:
In general I discourage super calls that hardcode a different parent. Skipping just one of the parent's methods implies that you want to use *some*, but not *all*, of the parent behavior.

This is more commonly addressed by re-examining the parent class hierarchy. In this case, the only thing you get for inheriting from TypeField is some arguments set by default in `__init__`.

Not a huge issue here, but be aware of these code smells.